clang/lib/3C/DeclRewriter.cpp

```diff
@@ -418,7 +418,12 @@ void DeclRewriter::rewriteFunctionDecl(FunctionDeclReplacement *N) {
 /*static*/ std::map<Decl *, Decl *> DeclRewriter::VDToRDMap;
 /*static*/ std::set<Decl *> DeclRewriter::InlineVarDecls;
 void DeclRewriter::detectInlineStruct(Decl *D, SourceManager &SM) {
-  if (RecordDecl *RD = dyn_cast<RecordDecl>(D)) {
+  RecordDecl *RD = dyn_cast<RecordDecl>(D);
+  if (RD != nullptr &&
+      // With -fms-extensions (default on Windows), Clang injects an implicit
+      // `struct _GUID` with an invalid location, which would cause an assertion
+      // failure in SM.isPointWithin below.
+      RD->getBeginLoc().isValid()) {
```

Comment:
Could this cause us to ignore inline structs in macros? Would that be a problem?

Reply:
I don't know.  Is there something better we can do here to not crash?  What is the consequence if we miss detecting an inline struct?

Reply:
On closer inspection, this doesn't seem to be an issue. The corresponding code in `ConstraintBuilder.cpp` has a similiar check, so presumably missing the check here was just a mistake. 

https://github.com/correctcomputation/checkedc-clang/blob/1b8b690bcbdfca754f7d2f495e56cbee437fce73/clang/lib/3C/ConstraintBuilder.cpp#L33-L36

Reply:
A `FullSourceLoc` is just a wrapper object that contains a `SourceLocation` and a reference to the `SourceManager`; there is no semantic difference.  Ideally we would be consistent; feel free to file an issue for cleanup.